### PR TITLE
Feature/add any right check

### PR DESCRIPTION
**Pull request by SolovyovDmitry**
Kernel: add new rights(reuse)
BrokerSupport: add new method for checking on any right

### CODE REVIEW

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs [hunk 1]
```diff
@@ -93,6 +96,43 @@ public async Task<bool> HasRightsAsync(Guid? userId, bool includeIsAdminCheck, p
       return result.Message.Body;
     }
 
+    /// <inheritdoc/>
+    public async Task<bool> HasAnyRightAsync(params int[] rightIds)
+    {
+      return await HasAnyRightAsync(null, true, rightIds);
+    }
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs:103]: may be add for HasAnyRightAsync overload like on lines 63-66?

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs [hunk 2]
```diff
@@ -21,25 +21,41 @@ public interface IAccessValidator
...
-    /// <returns>True, if there's a UserId-RightId pair for all requsted rights in the database. False otherwise.</returns>
+    /// <param name="rightIds">Ids of the rights.</param>
+    /// <returns>True, if there's a UserId-RightId pair for all requested rights in the database. False otherwise.</returns>
     Task<bool> HasRightsAsync(Guid? userId, bool includeIsAdminCheck, params int[] rightIds);
+
+    /// <summary>
+    /// Checks whether the user has any of specified rights.
+    /// </summary>
+    /// <param name="rightIds">Ids of the rights.</param>
+    /// <returns>True, if there's a UserId-RightId pair for all requested rights in the database. False otherwise.</returns>
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs]: not for all

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs [hunk 3]
```diff
@@ -21,25 +21,41 @@ public interface IAccessValidator
...
+    Task<bool> HasAnyRightAsync(params int[] rightIds);
+
+    /// <summary>
+    /// Checks whether the user has any of specified rights.
+    /// </summary>
+    /// <param name="userId">Id of the user.</param>
+    /// <param name="includeIsAdminCheck">If this is true, the method should be included in the check whether the user is an admin.</param>
+    /// <param name="rightIds">Ids of the rights.</param>
+    /// <returns>True, if there's a UserId-RightId pair for all requested rights in the database. False otherwise.</returns>
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs]: the same: not for all

File: src/Kernel/Constants/Rights.cs [hunk 4]
```diff
@@ -2,19 +2,19 @@
...
-  /// <param name="AddEditRemoveEmailsTemplates">Right grants admin rights to text template service</param>
-  /// <param name="AddEditRemoveDepartments">Right grants admin rights to department service, work with project department</param> 
-  /// <param name="AddEditRemoveNews">Right grants admin rights to news service</param> 
-  /// <param name="AddEditRemovePositions">Right grants admin rights to position service</param> 
-  /// <param name="AddEditRemoveTime">Right grants admin rights to time service</param> 
-  /// <param name="AddEditRemoveCompanies">Right grants admin rights to company, office services</param> 
-  /// <param name="AddEditRemoveCompanyData">Right grants admin rights to company service to work with company users, office workspaces/workspaceTypes, booking service</param> 
-  /// <param name="AddRemoveUsersRoles">Right grants admin rights to right service to work with users roles</param> 
-  /// <param name="AddEditRemoveWiki">Right grants admin rights to wiki service</param>
+  /// <field name="AddEditRemoveUsers">Right grants admin rights to user, family, education, skill, office users, achievement </field>
```
Comment (vorobevamv) [inline src/Kernel/Constants/Rights.cs]: may be stay 'param'?
https://learn.microsoft.com/ru-ru/dotnet/csharp/language-reference/xmldoc/recommended-tags
https://habr.com/ru/post/41514/

Comment (SolovyovDmitry) [reply src/Kernel/Constants/Rights.cs]: maybe value then?

File: src/Kernel/Constants/Rights.cs [hunk 5]
```diff
@@ -24,9 +24,9 @@ public static class Rights
     public const int AddEditRemoveNews = 5;
     public const int AddEditRemovePositions = 6;
     public const int AddEditRemoveTime = 7;
-    public const int AddEditRemoveHistories = 8; // can be reused
+    public const int AddEditRemoveOffices = 8;
     public const int AddEditRemoveCompanies = 9;
-    public const int AddRemoveDepartmentData = 10; // reuse the int, we do not need the right 
+    public const int AddEditRemoveBooking = 10;
```
Comment (ndoriomedov) [inline src/Kernel/Constants/Rights.cs:31]: i think we need to update these rights' localisations in rights service's db

**Review by ndoriomedov — APPROVED** [review]

File: src/Kernel/Constants/Rights.cs (hunk 4 above)
Comment (vorobevamv) [reply src/Kernel/Constants/Rights.cs]: 'value' - OK;
and, please, add descriptions for AddEditRemoveOffices and AddEditRemoveBooking

Comment (vorobevamv) [reply src/Kernel/Constants/Rights.cs]: and fix Companies+CompanyData if needed

File: src/Kernel/Constants/Rights.cs (hunk 5 above)
Comment (SolovyovDmitry) [reply src/Kernel/Constants/Rights.cs:31]: great remind

File: src/Kernel/Constants/Rights.cs [hunk 6]
```diff
@@ -2,19 +2,21 @@
...
+  /// <value name="AddEditRemoveProjects">Right grants admin rights to project, task services</value>
+  /// <value name="AddEditRemoveEmailsTemplates">Right grants admin rights to text template service</value>
+  /// <value name="AddEditRemoveDepartments">Right grants admin rights to department service, work with project department</value>
+  /// <value name="AddEditRemoveNews">Right grants admin rights to news service</value>
+  /// <value name="AddEditRemovePositions">Right grants admin rights to position service</value>
+  /// <value name="AddEditRemoveTime">Right grants admin rights to time service</value>
+  /// <value name="AddEditRemoveOffices">Right grants admin rights to office service</value>
+  /// <value name="AddEditRemoveCompanies">Right grants admin rights to company service</value>
+  /// <value name="AddEditRemoveBooking">Right grants admin rights to booking service</value>
+  /// <value name="AddEditRemoveCompanyData">Right grants admin rights to company service</value>
```
Comment (vorobevamv) [inline src/Kernel/Constants/Rights.cs]: Right grants admin rights to company service to work with company users

AddEditRemoveCompanies!=AddEditRemoveCompanyData

**Review by vorobevamv — APPROVED** [review]

**Pull request merged**